# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/socket/nsITransportSecurityInfo.idl
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 2
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e netwerk/socket/nsITransportSecurityInfo.idl
--- a/netwerk/socket/nsITransportSecurityInfo.idl	Mon Oct 21 19:33:02 2019 +0000
+++ b/netwerk/socket/nsITransportSecurityInfo.idl	Mon Oct 21 19:49:01 2019 +0000
@@ -7,7 +7,6 @@
 #include "nsISupports.idl"
 
 interface nsIX509Cert;
-interface nsIX509CertList;
 
 %{ C++
 namespace IPC {
@@ -34,12 +33,12 @@
     /**
      * If certificate verification failed, this will be the peer certificate
      * chain provided in the handshake, so it can be used for error reporting.
-     * If verification succeeded, this will be null.
+     * If verification succeeded, this will be empty.
      */
-    readonly attribute nsIX509CertList failedCertChain;
+    readonly attribute Array<nsIX509Cert> failedCertChain;
 
     readonly attribute nsIX509Cert serverCert;
-    readonly attribute nsIX509CertList succeededCertChain;
+    readonly attribute Array<nsIX509Cert> succeededCertChain;
 
     [must_use]
     readonly attribute ACString cipherName;